### PR TITLE
SOT-51: Write chapter4 of Proofs, Args and Zk

### PR DESCRIPTION
**Pull request by VanhGer**
# Check List

- [x] Did you check your grammar / reader proof your document using ChatGPT?
- [x] Don't forget to squash commits into meaningful chunks before merging
- [x] After merging this, please update your task status!
- [x] Did you verify that your math formulas are rendered correctly?

### CODE REVIEW

File: articles/proofs-arguments-and-zero-knowledge/attachments/compute_tildeW.png
Comment (sonntuet1997) [inline articles/proofs-arguments-and-zero-knowledge/attachments/compute_tildeW.png:1]: file_name

File: articles/proofs-arguments-and-zero-knowledge/attachments/cost_of_SAT.png
Comment (sonntuet1997) [inline articles/proofs-arguments-and-zero-knowledge/attachments/cost_of_SAT.png:1]: file_name

File: articles/proofs-arguments-and-zero-knowledge/chapter_4.md [hunk 1]
```diff
@@ -0,0 +1,284 @@
...
+Status: Done
+Level: "3"
+---
+# Chapter4 - Interactive Proofs
+
+ In this chapter, our primary focus is on the Sum-check protocol and GKR protocol. While other parts are not as crucial, it is important to read them to fully understand the Interactive Proof.
+
+## 4.1 The Sum-Check Protocol
+
+Given a $v$-variate polynomial $g$ defined over a finite field $F$. The purpose of of the sum-check protocol is for prover provide this sum for verifier: $$H := \sum_{b_{1} \in \{0, 1\}} ... \sum_{b_{v} \in \{0, 1\}} g(b_{1},...,b_v)$$
```
Comment (sonntuet1997) [inline articles/proofs-arguments-and-zero-knowledge/chapter_4.md:12]: Grammar :(

**Review by Draply — APPROVED** [review]

File: articles/proofs-arguments-and-zero-knowledge/attachments/the_cost_of_sat.png
Comment (sonntuet1997) [inline articles/proofs-arguments-and-zero-knowledge/attachments/the_cost_of_sat.png:1]: Convert to table

File: terms/sumcheck_protocol.md [hunk 2]
```diff
@@ -0,0 +1,42 @@
+# Sum-check Protocol
+
+Given a $v$-variate polynomial $g$ defined over a finite field $F$. The purpose of the sum-check protocol is for prover provide this
```
Comment (sonntuet1997) [inline terms/sumcheck_protocol.md:3]: prover(s) ?

File: terms/restriction.md
Comment (ktokto313) [inline terms/restriction.md:1]: $q(r)=q(5)$
$\iff q(r)=(-6) * 5^2-4 * 5+32=-138$

$f(l(r))= f(l(5)) = f(7, -6)$
$\iff f(l(r)) =3 * 7 * (-6)+2 * (-6)=-138$

make sure you have spaces around the * symbol so github won't mistake it as italic

File: articles/proofs-arguments-and-zero-knowledge/chapter_4.md [hunk 3]
```diff
@@ -0,0 +1,353 @@
...
+$(i,j), (i', j')$. $P$ can treat
+$(i', j')$ and $(i, j)$ as a single entity. There are only $n^2/2^k$ entities after $k$ variables have been bound, then the total work
+that $P$ does is: $$O(\sum^{2logn}_{k=1} n^2/2^k)= O(n^2)$$
+
+## 4.5 Applications of the Super-Efficient MATMULT IP
+
+### 4.5.1 For Counting Triangles
+
+Let $A$ is the adjacency matrix of a simple graph. The formula to compute number of triangles in the graph is:
+$$6\Delta = \sum_{i,j\in\{1,...,n\}}(A^2)_{i,j} \cdot A_{ij}$$
```
Comment (ktokto313) [inline articles/proofs-arguments-and-zero-knowledge/chapter_4.md:137]:
```suggestion
$$6\Delta = \sum _ {i,j\in\{1,...,n\}}(A^2) _ {i,j} \cdot A _ {ij}$$
```
space out so it won't mistake _ as italic

File: articles/proofs-arguments-and-zero-knowledge/chapter_4.md [hunk 4]
```diff
@@ -0,0 +1,353 @@
...
+   of  $\tilde W_{i+1}(b^∗)$ and $\tilde W_{i+1}(c^∗)$.
+   - $V$ chooses $r^∗ \in F$ at random and sets $r_{i+1} = l(r^∗)$ and $m_{i+1} \leftarrow q(r^*)$ (for details, see
+   [Section 4.5.2](chapter_4.md#4.5%20Applications%20of%20the%20Super-Efficient%20MATMULT%20IP#4.5.2%20Reducing%20Multiple%20Polynomial%20Evaluations%20to%20One)).
+   - To evaluate $\tilde{add_i}(r_i, b^*, c^*)$,  $\tilde{mult_i}(r_i, b^*, c^*)$, we can do it in polylogarithmic time. We will
+   discuss this in [Section 4.6.6](chapter_4.md#4.6.6%20Evaluating%20$%20tilde%20{add}_i$%20and%20$%20tilde%20{mult}_i$%20Efficiently).
+4. $V$ checks directly that $m_d = \tilde W_d(r_d)$ using [VSBW13](../../terms/lagrange_interpolation.md). Note that $\tilde W_d$ is
+the **MLE** of input $x$, and $V$ can compute it in $O(n)$, which $n$ is the size of the input.
+
+**Example**: Compute $\tilde W$ in a circuit over $F_5$ consisting entirely of multiplication gates
+![compute_w](../../terms/attachments/compute_tilde_w.png)
```
Comment (ktokto313) [inline articles/proofs-arguments-and-zero-knowledge/chapter_4.md:318]: image not found

**Review by ktokto313 — APPROVED** [review]
Just fix all the math formula

File: terms/restriction.md
Comment (VanhGer) [reply terms/restriction.md:1]: Côl

File: articles/proofs-arguments-and-zero-knowledge/chapter_4.md [hunk 5]
```diff
@@ -0,0 +1,239 @@
...
+So the protocol is:
+
+- $P$ sends the claimed output $d$ to $V$ and an $(i, j)$ such that  $(A + I)^{d-1}_{ij} = 0$
+- $V$ needs to check:
+  - all entries of $(A + I)^d$ are nonzero
+  - $(A + I)^{d-1}_{ij}$ is indeed zero
+
+The first task is accomplished by combining the **MATMULT** protocol with the [GKR protocol](../../terms/gkr_protocol.md).
+The second task is similarly accomplished using $O(logd)$ invocations of the **MATMULT** protocol - since $V$ only needs one entry of
+$(A + I)^{d-1}$, $P$ does not need to send the matrix in full, and the total communication here is just polylog$(n)$.
```
Comment (thuan2172001) [inline articles/proofs-arguments-and-zero-knowledge/chapter_4.md:227]: Cannot render this math expression: "polylog$(n)$"

**Review by thuan2172001 — APPROVED** [review]

File: articles/proofs-arguments-and-zero-knowledge/chapter_4.md [hunk 6]
```diff
@@ -0,0 +1,239 @@
+---
+Title: Interactive Proofs
+Status: Done
+Level: "3"
+---
+# Chapter 4 - Interactive Proofs
+
+In this chapter, our primary focus is on the Sum-check protocol and [GKR protocol](../../terms/gkr_protocol.md). While other
```
Comment (sonntuet1997) [inline articles/proofs-arguments-and-zero-knowledge/chapter_4.md:8]: Link sumcheck here

**Pull request merged**